cmd/installer/cli/install.go

```diff
@@ -0,0 +1,1007 @@
+	}
+
+	airgapBundleFlag, err := cmd.Flags().GetString("airgap-bundle")
+	if err != nil {
+		return nil, fmt.Errorf("unable to get airgap-bundle flag: %w", err)
+	}
+	if airgapBundleFlag != "" {
+		// update the k0s config to install with airgap
+		airgap.RemapHelm(provider, cfg)
+		airgap.SetAirgapConfig(cfg)
+	}
+	// This is necessary to install the previous version of k0s in e2e tests
+	// TODO: remove this once the previous version is > 1.29
+	unstructured, err := helpers.K0sClusterConfigTo129Compat(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert cluster config to 1.29 compat: %w", err)
+	}
+	data, err := k8syaml.Marshal(unstructured)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal config: %w", err)
+	}
+	if err := os.WriteFile(cfgpath, data, 0600); err != nil {
+		return nil, fmt.Errorf("unable to write config file: %w", err)
+	}
+	return cfg, nil
+}
+
+// applyUnsupportedOverrides applies overrides to the k0s configuration. Applies first the
+// overrides embedded into the binary and after the ones provided by the user (--overrides).
+// we first apply the k0s config override and then apply the built in overrides.
+func applyUnsupportedOverrides(cmd *cobra.Command, cfg *k0sconfig.ClusterConfig) (*k0sconfig.ClusterConfig, error) {
+	embcfg, err := release.GetEmbeddedClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get embedded cluster config: %w", err)
+	}
+	if embcfg != nil {
+		overrides := embcfg.Spec.UnsupportedOverrides.K0s
+		cfg, err = config.PatchK0sConfig(cfg, overrides)
+		if err != nil {
+			return nil, fmt.Errorf("unable to patch k0s config: %w", err)
+		}
+		cfg, err = config.ApplyBuiltInExtensionsOverrides(cfg, embcfg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to release built in overrides: %w", err)
+		}
+	}
+
+	overridesFlag, err := cmd.Flags().GetString("overrides")
+	if err != nil {
+		return nil, fmt.Errorf("unable to get overrides flag: %w", err)
+	}
+	eucfg, err := helpers.ParseEndUserConfig(overridesFlag)
+	if err != nil {
+		return nil, fmt.Errorf("unable to process overrides file: %w", err)
+	}
+	if eucfg != nil {
+		overrides := eucfg.Spec.UnsupportedOverrides.K0s
+		cfg, err = config.PatchK0sConfig(cfg, overrides)
+		if err != nil {
+			return nil, fmt.Errorf("unable to apply overrides: %w", err)
+		}
+		cfg, err = config.ApplyBuiltInExtensionsOverrides(cfg, eucfg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to end user built in overrides: %w", err)
+		}
+	}
+
+	return cfg, nil
+}
+
+// DeterminePodAndServiceCIDRS determines, based on the command line flags,
+// what are the pod and service CIDRs to be used for the cluster. If both
+// --pod-cidr and --service-cidr have been set, they are used. Otherwise,
+// the cidr flag is split into pod and service CIDRs.
+func DeterminePodAndServiceCIDRs(cmd *cobra.Command) (string, string, error) {
+	if cmd.Flags().Changed("pod-cidr") && cmd.Flags().Changed("service-cidr") {
```

Comment:
This needs to be `||`
```suggestion
	if cmd.Flags().Changed("pod-cidr") || cmd.Flags().Changed("service-cidr") {
```